ddp-workspace/projects/ddp-sdk/src/lib/services/internationalization/ngxTranslate.service.ts

```diff
@@ -1,17 +1,30 @@
 import { Injectable } from '@angular/core';
 import { TranslateService } from '@ngx-translate/core';
-import { Observable, of } from 'rxjs';
-import { mergeMap, merge } from 'rxjs/operators';
+import { merge, Observable, of } from 'rxjs';
+import { mergeMap } from 'rxjs/operators';
 
 @Injectable()
 export class NGXTranslateService {
     constructor(private translate: TranslateService) { }
 
-    public getTranslation(word: string | Array<string>): Observable<object | string> {
-        return of(null).pipe(
-            merge(this.translate.onLangChange),
-            merge(this.translate.onDefaultLangChange),
-            mergeMap(() => this.translate.get(word))
+    public getTranslation(word: string, interpolateParams?: object): Observable<string>;
+    public getTranslation(words: Array<string>, interpolateParams?: object): Observable<object>;
+
+    public getTranslation(word: string | Array<string>, interpolateParams?: object): Observable<object | string> {
+        return merge(
+            of(null),
+            this.translate.onLangChange,
+            this.translate.onDefaultLangChange).pipe(
+                mergeMap(() => this.translate.get(word, interpolateParams))
+        );
+    }
+
+    public getTranslationObject<T>(transKey: string): Observable<T> {
+        return merge(
+            of(null),
+            this.translate.onLangChange,
+            this.translate.onDefaultLangChange.asObservable()).pipe(
+                mergeMap(() => this.translate.get(transKey))
         );
```

Comment:
Found in the gallery component the one place where we just grab an object with the translation keys and values. We treat and type that as its own separate case.